grpc-server/hardware.go

```diff
@@ -316,3 +320,14 @@ func (s *server) validateHardwareData(ctx context.Context, hw *hardware.Hardware
 	}
 	return nil
 }
+
+func (s *server) normalizeHardwareData(hw *hardware.Hardware) {
```

Comment:
Do we want to make this a method? I was going to suggest/offer a test for this function and realized that the harness would include all the complexities of mocking a `server`.

Reply:
```patch
diff --git a/grpc-server/hardware.go b/grpc-server/hardware.go
index dffd492..1527439 100644
--- a/grpc-server/hardware.go
+++ b/grpc-server/hardware.go
@@ -40,7 +40,7 @@ func (s *server) Push(ctx context.Context, in *hardware.PushRequest) (*hardware.
 	}
 
 	// normalize data prior to storing in the database
-	s.normalizeHardwareData(hw)
+	normalizeHardwareData(hw)
 
 	// validate the hardware data to avoid duplicate mac address
 	err := s.validateHardwareData(ctx, hw)
@@ -321,7 +321,7 @@ func (s *server) validateHardwareData(ctx context.Context, hw *hardware.Hardware
 	return nil
 }
 
-func (s *server) normalizeHardwareData(hw *hardware.Hardware) {
+func normalizeHardwareData(hw *hardware.Hardware) {
 	// Ensure MAC is stored as lowercase
 	if hw != nil && hw.Network != nil {
 		for i := range hw.Network.Interfaces {
diff --git a/grpc-server/hardware_test.go b/grpc-server/hardware_test.go
new file mode 100644
index 0000000..01d3a6d
--- /dev/null
+++ b/grpc-server/hardware_test.go
@@ -0,0 +1,49 @@
+package grpcserver
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/tinkerbell/tink/protos/hardware"
+)
+
+func Test_normalizeHardwareData(t *testing.T) {
+	type args struct {
+		hw *hardware.Hardware
+	}
+	tests := []struct {
+		name string
+		got  args
+		want args
+	}{
+		{
+			name: "NilHardware",
+			got:  args{hw: nil},
+		},
+		{
+			name: "Uppercase",
+			got: args{hw: &hardware.Hardware{
+				Network: &hardware.Hardware_Network{
+					Interfaces: []*hardware.Hardware_Network_Interface{{
+						Dhcp: &hardware.Hardware_DHCP{Mac: "DE:AD:BE:EF"},
+					}},
+				},
+			},
+			},
+			want: args{hw: &hardware.Hardware{
+				Network: &hardware.Hardware_Network{
+					Interfaces: []*hardware.Hardware_Network_Interface{{
+						Dhcp: &hardware.Hardware_DHCP{Mac: "de:ad:be:ef"},
+					}},
+				},
+			},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			normalizeHardwareData(tt.got.hw)
+			assert.Equal(t, tt.want.hw, tt.got.hw)
+		})
+	}
+}
```

Reply:
Indeed, already switched it to a standalone function as part of creating the tests